### PR TITLE
Autotracing - Travel step extended

### PR DESCRIPTION
**Pull request by atoncetti**
Implementation of [#950](https://github.com/jshmrtn/hygeia/issues/950)

### CODE REVIEW

**Review by jeremyzahner — APPROVED** [review]
LGTM

**Comment by jeremyzahner** [comment]
@atoncetti Currently we would probably not merge this but focus on some other smaller things. Let's talk about that separately.

**Comment by maennchen** [comment]
@atoncetti When creating the last country transmission via the auto tracing problem, the problem should be automatically resolved.

File: lib/hygeia_web/live/risk_country_live/index.ex [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    |> apply_action(:validate)
+    |> case do
+      {:ok, selected_countries} ->
+        risk_countries =
+          selected_countries
+          |> Map.get(:countries, [])
+          |> Enum.filter(& &1.is_risk_country)
+          |> Enum.map(& &1.country)
+
+        Ecto.Multi.new()
```
Comment (maennchen) [inline lib/hygeia_web/live/risk_country_live/index.ex:107]: move this function into the Context. `patch_risk_countries` sounds like a good name.

File: lib/hygeia_web/live/risk_country_live/index.ex [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+  @impl Phoenix.LiveView
+  def handle_info({_type, %RiskCountry{}, _version}, socket) do
+    {:noreply, assign(socket, risk_countries: RiskCountryContext.list_risk_countries())}
+  end
+
+  def handle_info(_other, socket), do: {:noreply, socket}
+
+  defp get_countries do
+    all_countries = countries()
```
Comment (maennchen) [inline lib/hygeia_web/live/risk_country_live/index.ex:139]: How about a function `RiskCountryContext.list_risk_country_codes` that directly only queries for the codes?

File: priv/gettext/de/LC_MESSAGES/default.po [hunk 3]
```diff
@@ -1077,29 +1086,29 @@ msgstr "Datum"
...
+msgctxt "Elixir.Hygeia.AutoTracingContext.AutoTracing.Travel Field"
+msgid "Date of last departure"
+msgstr "Datum der letzten Ausreise"
+
+#, elixir-format
+msgctxt "Auto Tracing Resolve Problems"
 msgid "Date of last visit"
 msgstr "Datum des letzten Besuchs"
 
-#, elixir-format, ex-autogen
```
Comment (maennchen) [inline priv/gettext/de/LC_MESSAGES/default.po:1083]: That label needs to stay in

File: priv/repo/migrations/20220103114334_migrate_auto_tracing_travel.exs [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  @moduledoc false
+
+  use Hygeia, :migration
+
+  def up do
+    execute(
+      fn ->
+        :ok = run_authentication(repo(), origin: :migration, originator: :noone)
+      end,
+      &noop/0
```
Comment (maennchen) [inline priv/repo/migrations/20220103114334_migrate_auto_tracing_travel.exs:12]: You can either remove the second arg of `execute` or do a `change` function. (The second arg of `execute` is for the other direction.)

File: lib/hygeia_web/live/risk_country_live/index.ex (hunk 2 above)
Comment (atoncetti) [reply lib/hygeia_web/live/risk_country_live/index.ex:126]: the funciton (get_countries) generates a list of all countries and marks those that are risk country (by looking at the risk countries in the db). End result a list of non risk countries bundled with risk countries.

**Review by maennchen — APPROVED** [review]

**Pull request merged**